### PR TITLE
[YUNIKORN-2658]add nolint:funlen to long functions to supress the lint warnings 

### PR DESCRIPTION
**Pull request by rich7420**
### What is this PR for?

To add nolint:funlen to long functions to supress the lint warnings, when we use `make lint` 
like this 
image

### What type of PR is it?

- [x] - Improvement

### Todos
-  Improve other warn message

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2658

### How should this be tested?
in [Makefile](https://github.com/apache/yunikorn-k8shim/blob/master/Makefile#L294)  take off `--new-from-rev=$${headSHA}`
then run `make lint
`
### Screenshots (if appropriate)
NA
### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/cache/application_test.go [hunk 1]
```diff
@@ -559,6 +559,7 @@ func TestGetNonTerminatedTaskAlias(t *testing.T) {
 	task1 := NewTask(taskID1, app, context, pod1)
 	app.taskMap[taskID1] = task1
 	task1.sm.SetState(TaskStates().Pending)
+	//nolint:funlen
```
Comment (chia7712) [inline pkg/cache/application_test.go:562]: Could you please move this suppression to line#527?

**Review by chia7712 — COMMENTED** [review]
@rich7420 thanks for patch. Could you please fix following warnings also?
```
pkg/cache/application_test.go:753: Function 'TestTryReservePostRestart' is too long (123 > 120) (funlen)
func TestTryReservePostRestart(t *testing.T) {
pkg/cache/context_test.go:1485: Function 'TestAddApplicationsWithTags' is too long (145 > 120) (funlen)
func TestAddApplicationsWithTags(t *testing.T) {

```

File: pkg/cache/application_test.go (hunk 1 above)
Comment (rich7420) [reply pkg/cache/application_test.go:562]: OK!!

File: pkg/cache/task_state.go [hunk 2]
```diff
@@ -314,6 +314,7 @@ func TaskStates() *TStates {
 	return storeTaskStates
 }
 
+//nolint:funlen
 func newTaskState() *fsm.FSM {
```
Comment (chia7712) [inline pkg/cache/task_state.go:318]: this is production code, so maybe we can do a bit refactor? for example, we can add two helper methods to generate `eventDesc` and `callbacks`. With those helpers, `newTaskState` can be simplified to following style
```go
func newTaskState() *fsm.FSM {
	states := TaskStates()
	return fsm.NewFSM(states.New, eventDesc(states), callbacks(states))
}

```

Comment (rich7420) [reply pkg/cache/task_state.go:318]: Ok! no problem.

**Comment by chia7712** [comment]
@rich7420 please take a look at failed tests?

**Comment by rich7420** [comment]
OK, no problem

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**